packages/core/src/index.ts

```diff
@@ -1,37 +1,38 @@
 console.warn('REAtom still work in progress, do not use it in production')
 
-type NodeId = string
-
-type ActionType = string
-type ActionTypesDictionary = Record<ActionType, true>
+type Dictionary<T> = { [key: string]: T }
 type Node = {
-  id: NodeId
-  actionTypes: ActionTypesDictionary
+  id: string
+  actionTypes: Dictionary<true>
   // TODO: try to remove it
-  dependencies: DependenciesDictionary
-  stackWorker: (ctx: Ctx) => any
+  dependencies: Dictionary<Node>
+  stackWorker: StackWorker
 }
-type DependenciesDictionary = Record<NodeId, Node>
-type State = Record<string, any>
+type State = Dictionary<any>
+
+type ReduceFunction<TState> = <T>(
+  dependency: Unit<T>,
+  reducer: (state: TState, value: T) => TState,
+) => void
 
 const NODE = Symbol('@@REAtom/NODE')
 const assign = Object.assign
 
-function noop() {}
+function noop() { }
 
-export type ActionCreator<Payload = undefined, Type extends string = string> = {
+export type ActionCreator<TPayload = undefined, TType extends string = string> = {
```

Comment:
Why not `Payload` and `Type`?

Reply:
When generic types uses without prefix it may result with type overrides 
```suggestion
export declare function declareAtom<State>(
export declare function declareAtom<TState>(
```
[its example](https://github.com/artalar/reatom/pull/70/files#diff-55646d75364ce42cdb84abec1a1531ccR120) when we have type `State` declared in file and generic type `State` declared in function

Reply:
it's more fun when we get something like this:
```javascript
export declare function declareAtom<State extends State>(
```

Reply:
```
export declare function declareAtom<S extends State>(
```

Reply:
@Wroud see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L114

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L119

Reply:
it's good way too